internal/pkg/manifest/backend_svc.go

```diff
@@ -101,37 +102,38 @@ func (s *BackendService) MarshalBinary() ([]byte, error) {
 
 // DockerfilePath returns the image build path.
 func (s *BackendService) DockerfilePath() string {
-	return s.Image.Build
+	return aws.StringValue(s.Image.Build)
 }
 
 // ApplyEnv returns the service manifest with environment overrides.
 // If the environment passed in does not have any overrides then it returns itself.
-func (s *BackendService) ApplyEnv(envName string) *BackendService {
-	target, ok := s.Environments[envName]
+func (s BackendService) ApplyEnv(envName string) (*BackendService, error) {
```

Comment:
I think I changed this simply to avoid this method from accidentally change the receiver `BackendService` since this method should only use the receiver to create another overrided `BackendService`.

Reply:
It's probably fine to keep it as a value receiver. I'm not certain but my guess is for consistency, so that if a client initializes a `BackendService` by value they will only be able to use the `ApplyEnv` method, whereas if it's a pointer they'll be able to use everything. Ideally, once you initialize you should see everything I think 🤔

Reply:
> so that if a client initializes a BackendService by value they will only be able to use the ApplyEnv method

This is confusing but true 🤦‍♂️